pkg/ottl/expression.go

```diff
@@ -549,6 +550,107 @@ func (g StandardIntLikeGetter[K]) Get(ctx context.Context, tCtx K) (*int64, erro
 	return &result, nil
 }
 
+// ByteSliceLikeGetter is a Getter that returns []byte by converting the underlying value to an []byte if necessary
+type ByteSliceLikeGetter[K any] interface {
+	// Get retrieves []byte value.
+	// The expectation is that the underlying value is converted to []byte if possible.
+	// If the value cannot be converted to []byte, nil and an error are returned.
+	// If the value is nil, nil is returned without an error.
+	Get(ctx context.Context, tCtx K) ([]byte, error)
+}
+
+type StandardByteSliceLikeGetter[K any] struct {
+	Getter func(ctx context.Context, tCtx K) (any, error)
+}
+
+func (g StandardByteSliceLikeGetter[K]) Get(ctx context.Context, tCtx K) ([]byte, error) {
+	val, err := g.Getter(ctx, tCtx)
+	if err != nil {
+		return nil, fmt.Errorf("error getting value in %T: %w", g, err)
+	}
+	if val == nil {
+		return nil, nil
+	}
+	var result []byte
+	switch v := val.(type) {
+	case []byte:
+		result = v
+	case string:
+		result = []byte(v)
+	case float64:
+		result = float64ToBytes(v)
+	case int64:
+		result = int64ToBytes(v)
+	case bool:
+		if v {
+			result = []byte{1}
+		} else {
+			result = []byte{0}
+		}
```

Comment:
Is this a valid representation of a bool -> a []byte? if we did `string([]byte{0}))` is that meaningful?

Reply:
>Is this a valid representation of a bool -> a []byte?

I'm not sure there's a canonical representation here, but I think that's as close as we can reasonably get since a bool is essentially a bit that is either flipped on or off.

>if we did string([]byte{0})) is that meaningful?

It's not particularly meaningful since `0` is the UTF-8 code point for "null". Printed as hex though, it's `00` which I think is sufficient.

Reply:
I'm leaning more towards returning an error if there isn't a well-defined, agreed upon mapping of bool -> []byte.

Reply:
Apparently Go has it the same way in the [encoding/binary package](https://pkg.go.dev/encoding/binary#:~:text=Boolean%20values%20encode%20as%20one,with%20blank%20(_)%20field%20names.).

Reply:
Alright then we're good